soul-spring-boot-starter/soul-spring-boot-starter-plugin/soul-spring-boot-starter-plugin-httpclient/src/main/java/org/dromara/soul/springboot/starter/plugin/httpclient/HttpClientPluginConfiguration.java

```diff
@@ -110,17 +107,11 @@ public HttpClient httpClient(final HttpClientProperties properties) {
                                     .to(builder::nonProxyHosts);
                         });
                     }
-                    tcpClient = tcpClient.option(ChannelOption.SO_TIMEOUT, 5000);
+                    tcpClient = tcpClient.option(ChannelOption.CONNECT_TIMEOUT_MILLIS, 5000);
+//                        // The write and read timeouts are serving as generic socket idle state handlers.
                     tcpClient = tcpClient.doOnConnected(c -> {
-                        ChannelPipeline pipeline = c.channel().pipeline();
-                        Map<String, ChannelHandler> nameChannelHandlerMap = pipeline.toMap();
-                        // The write and read timeouts are serving as generic socket idle state handlers.
-                        if (!nameChannelHandlerMap.containsKey("write_timeout")) {
-                            pipeline.addFirst("write_timeout", new WriteTimeoutHandler(properties.getWriteTimeout(), TimeUnit.MILLISECONDS));
-                        }
-                        if (!nameChannelHandlerMap.containsKey("read_timeout")) {
-                            pipeline.addFirst("read_timeout", new ReadTimeoutHandler(properties.getReadTimeout(), TimeUnit.MILLISECONDS));
-                        }
+                        c.addHandlerLast(new WriteTimeoutHandler(properties.getWriteTimeout(), TimeUnit.MILLISECONDS));
+                        c.addHandlerLast(new ReadTimeoutHandler(properties.getReadTimeout(), TimeUnit.MILLISECONDS));
```

Comment:
Please make sure that [this exception](https://github.com/dromara/soul/issues/1209)  will not occur.

Reply:
yes, i had tested it.

Reply:
@naah69 What does this PR actually fix? It's `read_timeout/write_timeout` that doesn't work? BTW, what is the actual effect of `addHandlerLast ` and `addHandlerFirst`?

Reply:
yes,read_timeout/write_timeout doesn't work.it allways make error that is `io.netty.handler.timeout.ReadTimeoutException: null`,then the request will be 500.

from Reactor Netty 0.8 and Spring Framework 5.1,the handler will be set by below code.i fix it to make code work.

```
TcpClient tcpClient = TcpClient.create()
                 .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, 1000)
                 .doOnConnected(connection ->
                         connection.addHandlerLast(new ReadTimeoutHandler(10))
                                   .addHandlerLast(new WriteTimeoutHandler(10)));
WebClient webClient = WebClient.builder()
    .clientConnector(new ReactorClientHttpConnector(HttpClient.from(tcpClient)))
    .build();
```

Reply:
So, what is the returned data when timeout occurs?

Reply:
@naah69 Any progress on this?